app/src/actioncommands.cpp

```diff
@@ -684,25 +664,31 @@ Status ActionCommands::deleteCurrentLayer()
     return Status::OK;
 }
 
-bool ActionCommands::checkNewLayerName(QString s)
+QString ActionCommands::nameSuggest(QString s)
 {
     LayerManager* layerMgr = mEditor->layers();
-    bool b = true;
+    // if no layers: return 's'
+    if (layerMgr->count() == 0)
+    {
+        return s;
+    }
+    QVector<QString> sLayers;
+    // fill Vector with layer names
     for (int i = 0; i < layerMgr->count(); i++)
     {
-        if (layerMgr->getLayer(i)->name() == s)
-        {
-            b = false;
-        }
+        sLayers.append(layerMgr->getLayer(i)->name());
     }
-    return b;
-}
-
-void ActionCommands::uniqueNameMsg()
-{
-    QMessageBox msgBox;
-    msgBox.setText(tr("Filename must be unique and not empty"));
-    msgBox.exec();
+    // if 's' is not in list, then return 's'
+    if (!sLayers.contains(s))
+    {
+        return s;
+    }
+    int j = 2;
```

Comment:
Not a strict requirement but I find it to be a bad habit and generally dislike seeing one char variable names. when possible please name variables something that makes sense to the context. j for example could be named increment and string could be named "name".

it's a small thing but it improves code readability a lot.

Reply:
Whether it's a bad habit, I don't know, but I tend to use bool b, String s, and int i, j and k. 
Being aware that I work on a project with many contributors, I will of course try to apply to the coding style.